src/OpenLoco/GameCommands/ChangeLandMaterial.cpp

```diff
@@ -0,0 +1,65 @@
+#include "../Map/MapGenerator.h"
+#include "../Map/TileLoop.hpp"
+#include "../Map/TileManager.h"
+#include "../Objects/LandObject.h"
+#include "../Objects/ObjectManager.h"
+#include "../S5/S5.h"
+#include "GameCommands.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x00468EDD
+    static uint32_t changeLandMaterial(Map::Pos2 pointA, Map::Pos2 pointB, uint8_t landType, uint8_t flags)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Construction);
+        uint8_t x = (pointA.x + pointB.x) / 2 + 16;
+        uint8_t y = (pointA.y + pointB.y) / 2 + 16;
+        auto height = Map::TileManager::getHeight({ x, y });
+        GameCommands::setPosition({ x, y, height.landHeight });
+
+        Map::TilePosRangeView tileLoop{ { pointA }, { pointB } };
+        for (const auto& tilePos : tileLoop)
+        {
+            auto surface = Map::TileManager::get(tilePos).surface();
+            if (surface != nullptr)
```

Comment:
if you invert and do surface == nullptr continue then it will reduce indentation. Same with the flags. Tbh could move the flag check to L21 and skip the whole function.

Reply:
Moved the flag check outside the loop and inverted the `nullptr` check, thanks!